apps/app_manager.py

```diff
@@ -29,17 +29,19 @@ class AppManager(object):
      """
     ordering_cache = {}
 
-    def __init__(self, app_directory, i, o):
+    def __init__(self, app_directory, i, o, config={}):
```

Comment:
Mutable default arguments are dangerous and may lead to surprising behaviour. Prefer using `config=None` instead.
[Stackoverflow reference](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument)

Reply:
The `config` is never changed in the code, and on top of that, the AppManager's __init__ is only supposed to be called once. Nevertheless, you're right - will change it for future-proofing =)